rust-wasm-id-allocator/distributed-id-allocator/src/compressor.rs

```diff
@@ -157,6 +157,15 @@ impl IdCompressor {
         };
 
         let range_base_local = LocalId::from_generation_count(range_base_gen_count);
+        let range_base_stable = StableId::from(session_id) + range_base_local;
+        if self.uuid_space.range_can_collide(
```

Comment:
This implementation checks the theoretical maximum new-cluster span (in the worst-case condition where the current tail cluster is exactly full). The more-specific implementation for this would be to check against the actual span of the new cluster. This would mean checking in two different places (conditional, only one would be called) -- one at line 180, and one at line 206. However, range_can_collide() requires a borrow of sessions and so borrow checking prevents calling this during the lifetime of tail_cluster.

Reply:
I agree with the thinking. Can you add this in a comment above the call? Also, rename the range_can_collide to range_collides since from uuid-table perspective it collides.

Reply:
This approach also means the compressor isn't left in a bad state, which I love!